### ISSUE TITLE
Remove support for caps on `User.username`

### ISSUE THREAD

**Issue by ErnestoFGonzalez** [hunchat/hunchat-api#4]
Remove support for caps on `User.username`

**Comment by ErnestoFGonzalez** [comment]
The recommended approach, from [this Stack Overflow answer](https://stackoverflow.com/a/49181581), seems to be to create a custom `UsernameField` and apply `lower` to `value` in the `get_prep_value` method. As so:

```
class UsernameField(models.CharField):
    def __init__(self, *args, **kwargs):
        super(UsernameField, self).__init__(*args, **kwargs)

    def get_prep_value(self, value):
        return str(value).lower()

class User(models.Model):
    username = UsernameField(max_length=USERNAME_MAX_LENGTH)
    ...
```

Note: other answers in the subject (see [here](https://stackoverflow.com/a/19502304)) mention possible south migration issues.

**Comment by ErnestoFGonzalez** [comment]
Fixed in f4d71144384118b0f214278eee012e0136442fd6
